diff -rc zfs-0.6.1.orig/module/Makefile.in zfs-0.6.1/module/Makefile.in
*** zfs-0.6.1.orig/module/Makefile.in	2013-03-26 09:37:47.000000000 -0700
--- zfs-0.6.1/module/Makefile.in	2013-04-05 23:50:41.497876000 -0700
***************
*** 18,26 ****
  	@# installed devel headers, or they may be in the module
  	@# subdirectory when building against the spl source tree.
  	@if [ -f @SPL_OBJ@/@SPL_SYMBOLS@ ]; then \
! 		/bin/cp @SPL_OBJ@/@SPL_SYMBOLS@ .; \
  	elif [ -f @SPL_OBJ@/module/@SPL_SYMBOLS@ ]; then \
! 		/bin/cp @SPL_OBJ@/module/@SPL_SYMBOLS@ .; \
  	else \
  		echo -e "\n" \
  		"*** Missing spl symbols ensure you have built the spl:\n" \
--- 18,26 ----
  	@# installed devel headers, or they may be in the module
  	@# subdirectory when building against the spl source tree.
  	@if [ -f @SPL_OBJ@/@SPL_SYMBOLS@ ]; then \
! 		cp @SPL_OBJ@/@SPL_SYMBOLS@ .; \
  	elif [ -f @SPL_OBJ@/module/@SPL_SYMBOLS@ ]; then \
! 		cp @SPL_OBJ@/module/@SPL_SYMBOLS@ .; \
  	else \
  		echo -e "\n" \
  		"*** Missing spl symbols ensure you have built the spl:\n" \
***************
*** 28,33 ****
--- 28,35 ----
  		"*** - @SPL_OBJ@/module/@SPL_SYMBOLS@\n"; \
  		exit 1; \
  	fi
+ 	@# when copying a file out of the nix store, we need to make it writable again. 
+ 	chmod +w @SPL_SYMBOLS@
  	$(MAKE) -C @LINUX_OBJ@ SUBDIRS=`pwd` @KERNELMAKE_PARAMS@ CONFIG_ZFS=m $@
  
  clean:
***************
*** 42,62 ****
  modules_install:
  	@# Install the kernel modules
  	$(MAKE) -C @LINUX_OBJ@ SUBDIRS=`pwd` $@ \
! 		INSTALL_MOD_PATH=$(DESTDIR)$(INSTALL_MOD_PATH) \
  		INSTALL_MOD_DIR=$(INSTALL_MOD_DIR)
  	@# Remove extraneous build products when packaging
! 	kmoddir=$(DESTDIR)$(INSTALL_MOD_PATH)/lib/modules/@LINUX_VERSION@; \
  	if [ -n $$kmoddir ]; then \
  		find $$kmoddir -name 'modules.*' | xargs $(RM); \
  	fi
! 	sysmap=$(DESTDIR)$(INSTALL_MOD_PATH)/boot/System.map-@LINUX_VERSION@; \
  	if [ -f $$sysmap ]; then \
  		depmod -ae -F $$sysmap @LINUX_VERSION@; \
  	fi
  
  modules_uninstall:
  	@# Uninstall the kernel modules
! 	kmoddir=$(DESTDIR)$(INSTALL_MOD_PATH)/lib/modules/@LINUX_VERSION@
  	list='$(subdir-m)'; for subdir in $$list; do \
  		$(RM) -R $$kmoddir/$(INSTALL_MOD_DIR)/$$subdir; \
  	done
--- 44,64 ----
  modules_install:
  	@# Install the kernel modules
  	$(MAKE) -C @LINUX_OBJ@ SUBDIRS=`pwd` $@ \
! 		INSTALL_MOD_PATH=@prefix@/$(INSTALL_MOD_PATH) \
  		INSTALL_MOD_DIR=$(INSTALL_MOD_DIR)
  	@# Remove extraneous build products when packaging
! 	kmoddir=@prefix@/$(INSTALL_MOD_PATH)/lib/modules/@LINUX_VERSION@; \
  	if [ -n $$kmoddir ]; then \
  		find $$kmoddir -name 'modules.*' | xargs $(RM); \
  	fi
! 	sysmap=@prefix@/$(INSTALL_MOD_PATH)/boot/System.map-@LINUX_VERSION@; \
  	if [ -f $$sysmap ]; then \
  		depmod -ae -F $$sysmap @LINUX_VERSION@; \
  	fi
  
  modules_uninstall:
  	@# Uninstall the kernel modules
! 	kmoddir=@prefix@/$(INSTALL_MOD_PATH)/lib/modules/@LINUX_VERSION@
  	list='$(subdir-m)'; for subdir in $$list; do \
  		$(RM) -R $$kmoddir/$(INSTALL_MOD_DIR)/$$subdir; \
  	done
