diff --git a/extra-builtins.cc b/extra-builtins.cc
index c205fe7..1e2c561 100644
--- a/extra-builtins.cc
+++ b/extra-builtins.cc
@@ -22,25 +22,28 @@
 
 using namespace nix;
 
-static BaseSetting<Path> extraBuiltinsFile{
-    settings.nixConfDir + "/extra-builtins.nix",
-        "extra-builtins-file",
-        "The path to a nix expression defining extra expression-language level builtins."};
-
-static RegisterSetting rp(&extraBuiltinsFile);
+struct ExtraBuiltinsSettings : Config
+{
+	Setting<Path> extraBuiltinsFile{ this, settings.nixConfDir + "/extra-builtins.nix"
+		                       , "extra-builtins-file"
+			               , "The path to a nix expression defining extra expression-language level builtins."
+	};
+};
 
+ExtraBuiltinsSettings extraBuiltinsSettings;
+static GlobalConfig::Register rp(&extraBuiltinsSettings);
 
 static void extraBuiltins(EvalState & state, const Pos & _pos,
     Value ** _args, Value & v)
 {
     if (state.allowedPaths)
-        state.allowedPaths->insert(extraBuiltinsFile);
+        state.allowedPaths->insert(extraBuiltinsSettings.extraBuiltinsFile);
 
     try {
         auto fun = state.allocValue();
-        state.evalFile(extraBuiltinsFile, *fun);
+        state.evalFile(extraBuiltinsSettings.extraBuiltinsFile, *fun);
         Value * arg;
-        if (settings.enableNativeCode) {
+        if (evalSettings.enableNativeCode) {
             arg = state.baseEnv.values[0];
         } else {
             arg = state.allocValue();
